### ISSUE TITLE
Convert to formula class

### ISSUE THREAD

**Issue by zx8754** [wenjie2wang/clusrank#1]
Related SO post - https://stackoverflow.com/q/73136299/680068

Where `lm` works fine and `clusWilcox.test` fails. `lm` manual says:
> formula - an object of class "formula" (or one that can be coerced to that class)

In `clusWilcox.test` check if it is not class of _formula_, then coerce to _formula_ to avoid the error.

**Comment by wenjie2wang** [comment]
Thanks.  It looks like a good suggestion that can be easily applied.

**Comment by wenjie2wang** [comment]
In brief, this "issue" should be resolved by bd6102705e6d7a4adb34fce5818c85528c165707.  

To be more specific, the `clusWilcox.test()` is an S3 generic function, with an S3 method for `formula` objects. The `default` method will be called if the input `x` is not a `formula` object.  The `substitute(mpg ~ i, list(i = as.name(x))` produces a `language` object, and thus, the `default` method is called, which (of course) does not know how to handle the input properly.  I do not think it is an issue of the function.   But why does `lm()` work?  Notice that `lm()` is not a generic function.  The conversion from `language` to `formula` is actually done in `model.frame.default()`.

``` r
remotes::install_github("wenjie2wang/clusrank")
library(clusrank)
packageVersion("clusrank")
#> [1] '1.0.3'

## example data
df <- mtcars[, c("vs", "am", "mpg")]
names(df)
#> [1] "vs"  "am"  "mpg"
# [1] "vs"  "am"  "mpg"

varlist <- names(df)[1:2]
# [1] "vs" "am"

lapply(varlist, function(x) {
  clusWilcox.test(substitute(mpg ~ i, list(i = as.name(x))), data = df)
})
#> [[1]]
#> 
#>  Clustered Wilcoxon rank sum test using Rosner-Glynn-Lee method
#> 
#> data:  mpg; group: vs; (from df)
#> number of observations: 32;  number of clusters: 32
#> Z = 3.9342, p-value = 8.349e-05
#> alternative hypothesis: true difference in locations is not equal to 0
#> 
#> 
#> [[2]]
#> 
#>  Clustered Wilcoxon rank sum test using Rosner-Glynn-Lee method
#> 
#> data:  mpg; group: am; (from df)
#> number of observations: 32;  number of clusters: 32
#> Z = 3.1291, p-value = 0.001753
#> alternative hypothesis: true difference in locations is not equal to 0
```

<sup>Created on 2022-07-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
